malloc/malloc.c

```diff
@@ -13,13 +13,14 @@
 char *string_dup(char *src)
 {
     int length = string_length(src);
-    char *dup_string = malloc(length);
+    char *dup_string = malloc(length * sizeof(char));
     while (*src != '\0') {
         *dup_string = *src;
         src++;
+        dup_string++;
     }
-    dup_string[length] = '\0';
-    return dup_string;
+    *dup_string = '\0';
+    return dup_string - length;
```

Comment:
As I mentioned to you before, this is a really hacky way to get back to the beginning of the dup_string.  Make sure you understand the alternative way to this implementation